### ISSUE TITLE
 Invalid work group size

### ISSUE THREAD

**Issue by njuhang** [tensorflow/tensorflow#61373]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

tf 2.10.0

### Custom code

Yes

### OS platform and distribution

ubuntu 20.04

### Mobile device

Qualcomm SM6225

### Python version

3.8

### Bazel version

5.1.1

### GCC/compiler version

9.4.0

### CUDA/cuDNN version

no

### GPU model and memory

QUALCOMM Adreno(TM) QUALCOMM OpenCL C 2.0 Adreno(TM) 610

### Current behavior?

use tflite to invoke mobilnet-v2 or any model with op softmax on mobile device, will encounter an error:
TfLiteGpuDelegate Invoke: Failed to clEnqueueNDRangeKernel - Invalid work group size.

More details:
if INFERENCE_FORCE_FP16 is false,  no error occured. 
if work group size is smaller than 256, no error occured.

### Standalone code to reproduce the issue

```shell
same code as tflite invoke on android phone will reproduce this issue.
```

**Comment by pjpratik** [comment]
Hi @njuhang 

Can you please pull from master branch and let us know if you are facing same issue?

Work groups from the below are currently used in TFLite GPU and their stability is statistically proven. While
they do not necessarily result in peak optimal time across all parameters, they are reliable in giving top 10% performance
regardless of the convolution parameters.



Adreno GPU Model | CONV 2D | DEPTHWISE CONV
-- | -- | --
630 | (4, 8, 4) | (4, 4, 8)
540 | (8, 2, 2) | (8, 8, 2)
510 | (8, 4, 4) | (8, 4, 4)
509 | (8, 4, 8) | (8, 4, 2)
50X/4XX | (8, 4, 8) | (8, 4, 8)

Please check tis paper for reference: https://arxiv.org/abs/1907.01989

Thanks.

**Comment by njuhang** [comment]
I download benchmark file from https://storage.googleapis.com/tensorflow-nightly-public/prod/tensorflow/release/lite/tools/nightly/latest/android_aarch64_benchmark_model_plus_flex, then benchmark on my sm6225 device and got the same error.

**Comment by pjpratik** [comment]
@njuhang Thanks for the information.

As this [comment](https://github.com/tensorflow/tensorflow/issues/34064#issuecomment-551936622) suggests,

The max Local Work group size of each device is often 256. We may have to experiment with the workgroup sizes and it is observed that Adreno is more susceptible to workgroup changes. Change in group size will be affecting in the performance.

Thanks.

**Comment by njuhang** [comment]
I see it. But I think it will result some errors on some devices such as sm6225, and it need to be solved.

**Comment by pjpratik** [comment]
@njuhang Thanks for the information.

@pkgoogle Could you please look into this issue?

Thanks.

**Comment by pkgoogle** [comment]
Hi @njuhang, thanks for reporting the issue. Can you help me with more context. Are you using the flex version of android_aarch64_benchmark_model? Because your command doesn't have it but your download link does. Can you upload the .tflite model so we can replicate? Are you doing this in android studio? Can you show me the code around where you invoke the model? A toy project with just the offending code will be preferable so that we can reproduce it locally.
